### PR TITLE
Add filename to LMDB reader errors

### PR DESCRIPTION
**Pull request by JanuszL**
- when LMDB reader throws error source filename is provided as well

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
- adds more informative error logging to LMDB reader

#### What happend in this PR?
 - added LMDB file name to error logs
 - tested locally by examining error message

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/pipeline/operators/reader/loader/lmdb.h [hunk 1]
```diff
@@ -24,36 +24,38 @@
...
   do { \
-    DALI_ENFORCE(status == MDB_SUCCESS, "LMDB Error: " + string(mdb_strerror(status))); \
+    DALI_ENFORCE(status == MDB_SUCCESS, "LMDB Error: " + string(mdb_strerror(status)) + \
+                                        ", with file: " + filename); \
   } while (0)
 
 namespace lmdb {
-  inline bool SeekLMDB(MDB_cursor* cursor, MDB_cursor_op op, MDB_val* key, MDB_val *value) {
+  inline bool SeekLMDB(MDB_cursor* cursor, MDB_cursor_op op, MDB_val* key, MDB_val *value,
+                       string &filename) {
```
Comment (mzient) [inline dali/pipeline/operators/reader/loader/lmdb.h]: Why not:
```suggestion
                       const string &filename) {
```
?

File: dali/pipeline/operators/reader/loader/lmdb.h [hunk 2]
```diff
@@ -24,36 +24,38 @@
...
       return false;
     } else {
-      CHECK_LMDB(status);
+      CHECK_LMDB(status, filename);
       return true;
     }
   }
 
-  inline uint64_t LMDB_size(MDB_txn* txn, MDB_dbi dbi) {
+  inline uint64_t LMDB_size(MDB_txn* txn, MDB_dbi dbi, string &filename) {
```
Comment (mzient) [inline dali/pipeline/operators/reader/loader/lmdb.h]:
```suggestion
  inline uint64_t LMDB_size(MDB_txn* txn, MDB_dbi dbi, const string &filename) {
```

File: dali/pipeline/operators/reader/loader/lmdb.h [hunk 3]
```diff
@@ -24,36 +24,38 @@
...
     MDB_stat stat;
 
-    CHECK_LMDB(mdb_stat(txn, dbi, &stat));
+    CHECK_LMDB(mdb_stat(txn, dbi, &stat), filename);
 
     return stat.ms_entries;
   }
 
-  inline void PrintLMDBStats(MDB_txn* txn, MDB_dbi dbi) {
+  inline void PrintLMDBStats(MDB_txn* txn, MDB_dbi dbi, string &filename) {
```
Comment (mzient) [inline dali/pipeline/operators/reader/loader/lmdb.h]:
```suggestion
  inline void PrintLMDBStats(MDB_txn* txn, MDB_dbi dbi, const string &filename) {
```

Comment (JanuszL) [reply dali/pipeline/operators/reader/loader/lmdb.h]: Done

File: dali/pipeline/operators/reader/loader/lmdb.h (hunk 2 above)
Comment (JanuszL) [reply dali/pipeline/operators/reader/loader/lmdb.h]: Done

File: dali/pipeline/operators/reader/loader/lmdb.h (hunk 1 above)
Comment (JanuszL) [reply dali/pipeline/operators/reader/loader/lmdb.h]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**